stl/src/filesystem.cpp

```diff
@@ -789,18 +789,21 @@ struct __std_fs_file_id { // typedef struct _FILE_ID_INFO {
 namespace {
     _Success_(return > 0 && return < nBufferLength) DWORD WINAPI _Stl_GetTempPath2W(
         _In_ DWORD nBufferLength, _Out_writes_to_opt_(nBufferLength, return +1) LPWSTR lpBuffer) noexcept {
+#if !defined(_ONECORE)
         // See GH-3011: This is intentionally not attempting to cache the function pointer.
         // TRANSITION, ABI: This should use __crtGetTempPath2W after this code is moved into the STL's DLL.
-        using _Fun_ptr = decltype(&::GetTempPath2W);
 
+        // use GetTempPath2W if it is available (only on Windows 11+)...
         const auto _Kernel32 = ::GetModuleHandleW(L"kernel32.dll");
         _Analysis_assume_(_Kernel32);
-        _Fun_ptr _PfGetTempPath2W = reinterpret_cast<_Fun_ptr>(::GetProcAddress(_Kernel32, "GetTempPath2W"));
-        if (!_PfGetTempPath2W) {
-            _PfGetTempPath2W = &::GetTempPathW;
+        const auto _Pf = reinterpret_cast<decltype(&::GetTempPath2W)>(::GetProcAddress(_Kernel32, "GetTempPath2W"));
+        if (_Pf) {
+            return _Pf(nBufferLength, lpBuffer);
         }
+#endif // ^^^ !defined(_ONECORE) ^^^
 
-        return _PfGetTempPath2W(nBufferLength, lpBuffer);
+        // ...otherwise use GetTempPathW.

```

Comment:
non-blocking nit: I'm not sure if this is a pre-existing convention in the codebase, but I'm not the biggest fan of "split comments" separated by "...". As the code changes, it can get lost that these were part of the same sentence.

In any case, this is a subjective stylistic nit. The change seems correct, thank you!

Reply:
Yeah, they're pretty rare, we probably ought to make them rarer. At least the ellipses make it somewhat clear that they're partial (I absolutely agree that they would be abominations without ellipses).